upload-api/tables/index.js

```diff
@@ -26,3 +26,55 @@ export const uploadTableProps = {
   // space + root must be unique to satisfy index constraint
   primaryIndex: { partitionKey: 'space', sortKey: 'root' },
 }
+
+/** @type TableProps */
+export const delegationTableProps = {
+  fields: {
+    link: 'string',        // `baf...x`
+    audience: 'string',   // `did:web:service`
+    issuer: 'string',     // `did:key:agent`
+    expiration: 'string', // `9256939505` (unix timestamp)
+    insertedAt: 'string', // `2022-12-24T...`
+    updatedAt: 'string',  // `2022-12-24T...`
+  },
+  // TODO does this index setup seem right?
+  // we want to query by audience, but that won't necessarily be unique, so use cid as sortKey
+  primaryIndex: { partitionKey: 'audience', sortKey: 'link' },
+}
+
+/** @type TableProps */
+export const subscriptionTableProps = {
+  fields: {
+    cause: 'string',        // `baf...x` (CID of invocation that created this subscription)
+    provider: 'string',   // `did:web:service` (DID of the provider, e.g. a storage provider)
+    customer: 'string',   // `did:mailto:agent` (DID of the user account)
+    subscription: 'string',      // string (arbitrary string associated with this subscription)
+    insertedAt: 'string', // `2022-12-24T...`
+    updatedAt: 'string',  // `2022-12-24T...`
+  },
+  // TODO does this index setup seem right?
+  primaryIndex: { partitionKey: 'customer', sortKey: 'subscription' },
+  globalIndexes: {
+    // TODO: I don't think we should keep this index - partitioning by provider is basically useless and won't be much faster than a table scan
+    provider: { partitionKey: 'provider', projection: ['customer'] },
```

Comment:
I'm no dynamo expert, we should ask @mikeal for the input. However I would expect that providers may want to have a view over their customers with a good enough query ability.

Reply:
yea agree - I'm just not sure that maintaining an index makes sense since we'll practically be doing a table scan every time... I think this is a Dynamo best-practices kind of question so yea would definitely love to get @mikeal's brain on this if we can!